MAGETWO-89081: CE edition - Update composer dependencies

joanhe opened the pull request:
### Fixed Issues 
* [MAGETWO-89081](https://jira.corp.magento.com/browse/MAGETWO-89081) CE edition - Update composer dependencies

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
Update colinmollenhour/php-redis-session-abstract

* Update implementation of `Cm\RedisSession\Handler\ConfigInterface`
* Add Redis Sentinel config options to `setup:config:set` command

Commit:
Update/remove packages in composer.json

allows compoer install on php71 and php72

Commit:
Remove use of depracated TableHelper

Replace references of `Symfony\Console\Helper\TableHelper` with
`Symfony\Console\Helper\Table`. `new` was used instead of DI to mirror
existing reference in
https://github.com/magento/magento2/blob/a083717504f63fd22748fb42d74b63c2906226ef/app/code/Magento/Developer/Console/Command/DiInfoCommand.php#L84
Not sure which is preferred.

Commit:
Fix reference to removed Diaglog console helper

It is replaced with `Symfony\Console\Helper\QuestionHelper`

Commit:
Remove mocking of Final class

`Symfony\Component\Console\Helper\ProgressBar` has been declared Final
and cannot be mocked.

Commit:
Correct default argument type in MultipleStreamOutput

The second constructor argument of parent class
`Symfony\Component\Console\Output\Output` must be boolean.

Commit:
Correct comparison logic in ConfigSetCommandTest

Commit:
Move libraries back to require-dev

Commit:
Update PHP version requirements in all composer.json

Update to "~7.1.3||~7.2.0"

Commit:
magento/framework composer.json updates

Match library versions with those in project composer.json

Commit:
Replace php 7.0 with 7.2 in .travis.yml

Fixes #25

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-89081

# Conflicts:
#	app/code/Magento/SalesSequence/composer.json
#	composer.json
#	composer.lock
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Catalog/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/CatalogSearch/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Checkout/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/ConfigurableProduct/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/SalesRule/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Search/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Theme/composer.json
#	dev/tests/acceptance/tests/functional/Magento/FunctionalTest/Wishlist/composer.json
#	lib/internal/Magento/Framework/composer.json

Commit:
MAGETWO-89081: CE edition - Update composer dependencies, and fix Travis build issues

- revert changes in e20d5b8f024eba4010d6d561a31100838316cccf in function tests

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

- fix unit test failures

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

- update colinmollenhour/cache-backend-file to 1.4.1

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

- update zendframework/zend-code to 3.1.0 and zendframework/zend-mail to 2.9.0

buskamuza commented:
✅ Changes look good
@joanhe , feel free to merge after changes in tests caused by `zend-code` updates are reverted

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

- revert change to code generater test after update zendframework/zend-code to 3.1.0

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

- fix count() issue

Commit:
MAGETWO-89091: CE edition - Update composer dependencies

- change magento/zendframework1 to dev-master
